pycontrails/datalib/goes.py

```diff
@@ -564,6 +564,14 @@ def _get_without_cache(self, time: datetime.datetime) -> xr.DataArray:
         return da
 
 
+def _load_via_tempfile(data: bytes) -> xr.Dataset:
+    """Load xarray dataset via temporary file."""
+    with tempfile.NamedTemporaryFile(buffering=0) as tmp:
+        tmp.write(data)
+        ds = xr.load_dataset(tmp.name)
+    return ds
```

Comment:
You can save a single line of code with:

```python
    with tempfile.NamedTemporaryFile(buffering=0) as tmp:
        tmp.write(data)
        return xr.load_dataset(tmp.name)
```

(the context manager will still exit even if there's a return in there)

Reply:
https://github.com/contrailcirrus/pycontrails/pull/213/commits/44911214d71a9af2b39043f84ec6076cdccbf6ac